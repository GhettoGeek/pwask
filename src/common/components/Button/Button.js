import React from 'react'
import PropTypes from 'prop-types'
import Button from '@material-ui/core/Button'

function EnhancedButton({
  variant, color, children, ...props
}) {
  return (
    <Button
      variant={variant}
      color={color}
      {...props}
    >
      {children}
    </Button>
  )
}

EnhancedButton.propTypes = {
  variant: PropTypes.string,
  color: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape(),
  ]),
}

EnhancedButton.defaultProps = {
  variant: 'contained',
  color: 'primary',
  children: '',
}

export default EnhancedButton
