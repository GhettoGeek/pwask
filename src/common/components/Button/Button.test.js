import React from 'react'
import { shallow } from 'enzyme'
import Button from './Button'

describe('Button', () => {
  it('renders correctly the component', () => {
    const wrapper = shallow(<Button />)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
