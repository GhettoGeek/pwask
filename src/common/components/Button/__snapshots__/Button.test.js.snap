// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly the component 1`] = `"<button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\"></span></button>"`;
